lib/service/database.js

```diff
@@ -284,6 +288,12 @@ DatabaseService.prototype.queueAnalysisOperations = function(analysis, callback)
                 asyncQueries.push({
                     query: updateNodeStatusAtAnalysisCatalogQuery(nodeIds, Node.STATUS.RUNNING)
                 });
+                if (sqlWrappedNode.preCheckNodeQuery() !== undefined) {
+                    asyncQueries.push({
+                        query: readOnlyTransactionQuery([sqlWrappedNode.preCheckNodeQuery()]),
```

Comment:
[This](https://github.com/CartoDB/camshaft/pull/317/files#diff-05ff3cd3ac12bad795f26bd25a01be43R152) test should check that the pre-check query is correctly generated including the read only part.